transport/src/main/java/io/netty/channel/embedded/EmbeddedChannel.java

```diff
@@ -338,6 +351,17 @@ public boolean writeInbound(Object... msgs) {
             return isNotEmpty(inboundMessages);
         }
 
+        if (!config().isAutoRead()) {
+            Queue<AbstractMap.SimpleEntry<Object, ChannelPromise>> tempInboundMessages = tempInboundMessages();
+            if (tempInboundMessages != null) {
+                assert tempInboundMessages.size() == 0;
```

Comment:
I dont think this `assert` is correct. It will fail if you call `writeInbound` multiple times while autoRead is false.